core/templates/dev/head/services/AudioPlayerService.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to operate the playback of audio.
+ */
+
+oppia.factory('AudioPlayerService', [
+  'ngAudio', 'AssetsBackendApiService', 'explorationContextService',
+  function(ngAudio, AssetsBackendApiService, explorationContextService) {
+    var _currentTrackFilename;
+    var _currentTrack;
+
+    var _play = function(filename) {
+      AssetsBackendApiService.loadAudio(
+        explorationContextService.getExplorationId(), filename)
+        .then(function(audioBlob) {
+          var blobUrl = URL.createObjectURL(audioBlob);
+          _currentTrack = ngAudio.load(blobUrl);
+          _currentTrackFilename = filename; 
+          _currentTrack.play();
+        });
+    };
+
+    var _pause = function() {
+      if (_currentTrack) {
+        _currentTrack._pausee();
```

Comment:
oops, was originally planning to remove pause and delay all the other audio functions cuz i thought it might be easier to test later. but i changed my mind. it all works now